homeassistant/components/notify/__init__.py

```diff
@@ -50,22 +57,88 @@
 )
 
 
+@bind_hass
+async def async_reload(hass, integration_name):
+    """Register notify services for an integration."""
+    if NOTIFY_SERVICES not in hass.data:
+        return
+
+    data = hass.data[NOTIFY_SERVICES][integration_name]
+    notify_service = data[SERVICE]
+    friendly_name = data[FRIENDLY_NAME]
+    targets = data[TARGETS]
+
+    async def _async_notify_message(service):
+        """Handle sending notification message service calls."""
+        await _async_notify_message_service(hass, service, notify_service, targets)
+
+    if hasattr(notify_service, "targets"):
+        for name, target in notify_service.targets.items():
```

Comment:
For a followup PR, I was thinking about removing services that are no longer present so we can reload again when we remove a mobile_app